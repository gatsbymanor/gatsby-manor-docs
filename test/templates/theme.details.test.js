import React from 'react';
import ThemeDetailsTemplate from '../../src/templates/theme.details';

describe('<ThemeDetailsTemplate />', () => {
  const testData = {
    massivelyImage: 'image',
    lensImage: 'image',
    photonImage: 'image',
    identityImage: 'image',
    dimensionImage: 'image',
    tessellateImage: 'image',
    location: {
      pathname: "/home",
    },
    markdownRemark: {
      html: {},
      frontmatter: {},
    },
  };

  const spy = jest.spyOn(ThemeDetailsTemplate.prototype, 'toggleSidemenu');

  const component = shallow(
    <ThemeDetailsTemplate
      data={testData}
      location={testData.location}/>
  );

  it('should render', () => {
    expect(component).toMatchSnapshot();
  });

  it('should toggle the sidemenu on menu button click', () => {
    const openSideMenuButtion = component.find('.theme--details--main--open');
    openSideMenuButtion.simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('should toggle the sidemenu on close button click', () => {
    const closeSideMenuButtion = component.find('.theme--details--sidemenu--close');
    closeSideMenuButtion.simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('should have the right props', () => {
    const props = component.instance().props;

    expect(props).toHaveProperty('data');
    expect(props).toHaveProperty('data.markdownRemark');
    expect(props).toHaveProperty('data.markdownRemark.html');
    expect(props).toHaveProperty('data.markdownRemark.frontmatter');

    expect(props).toHaveProperty('data.massivelyImage');
    expect(props).toHaveProperty('data.lensImage');
    expect(props).toHaveProperty('data.photonImage');
    expect(props).toHaveProperty('data.identityImage');
    expect(props).toHaveProperty('data.dimensionImage');
    expect(props).toHaveProperty('data.tessellateImage');

    expect(props).toHaveProperty('location');
    expect(props).toHaveProperty('location.pathname');

  });
});
