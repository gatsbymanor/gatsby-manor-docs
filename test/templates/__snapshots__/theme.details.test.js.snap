// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeDetailsTemplate /> should render 1`] = `
<div
  className="theme--details"
>
  <div
    className="theme--details--column theme--details--main"
  >
    <div
      className="theme--details--main--open"
      onClick={[Function]}
    >
      ☰
    </div>
    <div
      className="theme--details--img--small"
    >
      <Image
        alt="title theme"
        fadeIn={true}
        sizes={null}
        title="Theme details image"
      />
    </div>
    <div
      className="theme--details--img--large"
    >
      <Image
        alt="title theme"
        fadeIn={true}
        sizes={null}
        title="Theme details image"
      />
    </div>
  </div>
  <div
    className="theme--details--column theme--details--sidemenu"
  >
    <div
      className="theme--details--sidemenu--close"
      onClick={[Function]}
    >
      ×
    </div>
    <h1
      className="theme--details--sidemenu--title"
    />
    <div
      className=""
      dangerouslySetInnerHTML={
        Object {
          "__html": Object {},
        }
      }
    />
  </div>
</div>
`;
